trin-storage/src/config.rs

```diff
@@ -0,0 +1,81 @@
+use std::path::PathBuf;
+
+use discv5::enr::NodeId;
+use ethportal_api::types::cli::{BEACON_NETWORK, HISTORY_NETWORK, STATE_NETWORK};
+use r2d2::Pool;
+use r2d2_sqlite::SqliteConnectionManager;
+
+use crate::{error::ContentStoreError, utils::setup_sql, DistanceFunction};
+
+const BYTES_IN_MB_U64: u64 = 1000 * 1000;
+
+/// Factory for creating [PortalStorageConfig] instances
+pub struct PortalStorageConfigFactory {
+    node_id: NodeId,
+    node_data_dir: PathBuf,
+    total_capacity_mb: u64,
+    total_capacity_weight: u64,
+    sql_connection_pool: Pool<SqliteConnectionManager>,
+}
+
+impl PortalStorageConfigFactory {
+    const HISTORY_CAPACITY_WEIGHT: u64 = 1;
```

Comment:
Re: 99/1 (state/history capacity)....
I guess I don't have any good, sound reasoning for why it shouldn't be that, just that this seems a little high... But, it is true that the network has much higher storage demands for state than history data, and as long as we don't update our current mainnet 4444s nodes to support state network, then 99/1 is fine by me....

Though, I wonder if we want to introduce some concept of a minimum (total) capacity? Idk, like `100mb`? (though, the only repercussion I can think of is that this minimum would break [this test](https://github.com/ethereum/trin/blob/master/ethportal-peertest/src/scenarios/gossip.rs#L298))

Reply:
We should definitely not enable state network on the 4444 nodes. That would be bad regardless of the ratio that we pick here.
With that in mind, I will leave the ratio as 99/1 for now (we can easily change it if there is need for it).

I don't have strong opinion about enforcing minimum capacity. It's something we can discuss in the next meeting. But I would leave it out of this pr.